### ISSUE TITLE
FR: Trigger on qualified macros as well

### ISSUE THREAD

**Issue by lgoettgens** [LilithHafner/BasicAutoloads.jl#10]
I often use `Profile.Allocs.@profile` without `using Profile` beforehand. I suppose this needs some small tweak in this package somewhere, to make the following part of my `startup.jl` work.

```julia
BasicAutoloads.register_autoloads([
        ["@profile", "Profile.Allocs.@profile"] => :(using Profile, PProf),
    ])
```

**Comment by LilithHafner** [comment]
Ah, this is actually a bug:

```julia-repl
julia> BasicAutoloads.register_autoloads([
               ["foo"] => :(println("Foo!")),
               ["bar"] => :(println("Bar!")),
           ])

julia> foo
Foo!
ERROR: UndefVarError: `foo` not defined in `Main`
Suggestion: check for spelling errors or missing imports.

julia> x.bar
ERROR: UndefVarError: `x` not defined in `Main`
Suggestion: check for spelling errors or missing imports.
Stacktrace:
 [1] top-level scope
   @ REPL[7]:1
```

Entering `Profile.Allocs.@profile` in the REPL should but does not trigger autoloads for "@profile".

**Comment by LilithHafner** [comment]
1.
This is fixed in BasicAutoloads 1.0.2. Thanks for reporting!

2.
You only need
```julia
BasicAutoloads.register_autoloads([
        ["@profile"] => :(using Profile, PProf),
    ])
```
the `"Profile.Allocs.@profile"` trigger will only match `var"Profile.Allocs.@profile", not `Profile.Allocs.@profile`. As per the docstring of `BasicAutoloads.register_autoloads`
> Each trigger must be a String. If the trigger is found as a symbol (e.g. variable, function, or macro name) in an input expression to the REPL...
